### PR TITLE
Add a way to plug device-discoveries

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Provide a way for extensions to plug their own `IDeviceDiscovery` instances. 
`$devicesService` will call `startLookingForDevices` for every one of them and make sure to attach to `deviceLost`/`deviceFound` events

Ping @rosen-vladimirov @TsvetanMilanov @yyosifov

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/mobile-core/devices-service.ts [hunk 1]
```diff
@@ -234,13 +218,23 @@ export class DevicesService implements Mobile.IDevicesService {
...
 			await this.detectCurrentlyAttachedDevices();
 			await this.startDeviceDetectionInterval();
-		} else if (this.$mobileHelper.isiOSPlatform(this._platform)) {
-			await this.$iOSDeviceDiscovery.startLookingForDevices();
-			if (this.$hostInfo.isDarwin) {
-				await this.$iOSSimulatorDiscovery.startLookingForDevices();
+		} else {
+			if (this.$mobileHelper.isiOSPlatform(this._platform)) {
+				await this.$iOSDeviceDiscovery.startLookingForDevices();
+				if (this.$hostInfo.isDarwin) {
```
Comment (rosen-vladimirov) [inline mobile/mobile-core/devices-service.ts]: I believe you do not need this anymore

File: test/unit-tests/mobile/devices-service.ts [hunk 2]
```diff
@@ -256,8 +269,19 @@ describe("devicesService", () => {
...
+		let devices = devicesService.getDeviceInstances();
+		assert.isTrue(devicesService.hasDevices, "After emitting two devices, hasDevices must be true");
+		assert.deepEqual(devices[0], iOSDevice);
+		assert.deepEqual(devices[1], androidDevice);
+	});
+
+	it("attaches to events when a new custom DevicesService is instantiated", () => {
+		const customDeviceDiscovery = testInjector.resolve(CustomDeviceDiscoveryStub);
+		devicesService.addDeviceDiscovery(customDeviceDiscovery);
+		customDeviceDiscovery.emit(DeviceDiscoveryEventNames.DEVICE_FOUND, iOSDevice);
```
Comment (rosen-vladimirov) [inline test/unit-tests/mobile/devices-service.ts]: can you add assert before emitting that there are no devices at the moment.

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**